### ISSUE TITLE
Trap in FGUI while remote connected - no problem for Injoy usage but maybe libcn0.dll regression

### ISSUE THREAD

**Issue by AndiB-2** [bitwiseworks/InJoy#61]
------------------------------------------------------------

07-28-2019  11:31:09  SYS3171  PID 00b5  TID 0004  Slot 0106
P:\INJOYFIREWALL4.2.2\FGUI.EXE
c0000005
19fb28d4
P1=00000002  P2=026afc7c  P3=XXXXXXXX  P4=XXXXXXXX  
EAX=0000150b  EBX=026b005c  ECX=026b051c  EDX=026afc7c
ESI=026b0724  EDI=026b032c  
DS=0053  DSACC=f0f3  DSLIM=ffffffff  
ES=0053  ESACC=f0f3  ESLIM=ffffffff  
FS=150b  FSACC=00f3  FSLIM=00000030
GS=0000  GSACC=****  GSLIM=********
CS:EIP=005b:19f1642a  CSACC=f0df  CSLIM=ffffffff
SS:ESP=0053:026b0000  SSACC=f0f3  SSLIM=ffffffff
EBP=026b0088  FLG=00010206

LIBCN0.DLL 0001:000e28d4

**To Reproduce**
1. remote connect to server running the firewall
2. Click on File - Restart firewall server
3. confirm firewall restart
4. --> popuplog.os2 entry see above

This is with current netlabs-rel libc version. Currently not tested with -exp repository.

**Comment by SilvanScherrer** [comment]
if this is still an issue, you need to install the debug version. As only then we see what might cause it.

**Comment by AndiB-2** [comment]
Yes. Tested on another system which at the first test crashes rock solid - no mouse no CTRL-ALT-DEL. Second trial without hang but popuplog too. popuplog is a bit different - 
------------------------------------------------------------

11-09-2020  20:11:33  SYS3171  PID 00ff  TID 0004  Slot 00c4
P:\INJOYFIREWALL4.2.2\FGUI.EXE
c0000005
19facc54
P1=00000002  P2=026afc5c  P3=XXXXXXXX  P4=XXXXXXXX  
EAX=026b005c  EBX=026b005c  ECX=19eff966  EDX=00000014
ESI=19eff966  EDI=026aff90  
DS=0053  DSACC=d0f3  DSLIM=bfffffff  
ES=0053  ESACC=d0f3  ESLIM=bfffffff  
FS=150b  FSACC=00f3  FSLIM=00000030
GS=0000  GSACC=****  GSLIM=********
CS:EIP=005b:19efd718  CSACC=d0df  CSLIM=bfffffff
SS:ESP=0053:026afe20  SSACC=d0f3  SSLIM=bfffffff
EBP=026b0018  FLG=00010202

LIBCN0.DLL 0001:000fcc54

------------------------------------------------------------

11-09-2020  20:19:41  SYS3171  PID 00b5  TID 0004  Slot 00b2
P:\INJOYFIREWALL4.2.2\FGUI.EXE
c0000005
1a03cc54
P1=00000002  P2=026afc5c  P3=XXXXXXXX  P4=XXXXXXXX  
EAX=026b005c  EBX=026b005c  ECX=19f8f966  EDX=00000014
ESI=19f8f966  EDI=026aff90  
DS=0053  DSACC=d0f3  DSLIM=bfffffff  
ES=0053  ESACC=d0f3  ESLIM=bfffffff  
FS=150b  FSACC=00f3  FSLIM=00000030
GS=0000  GSACC=****  GSLIM=********
CS:EIP=005b:19f8d718  CSACC=d0df  CSLIM=bfffffff
SS:ESP=0053:026afe20  SSACC=d0f3  SSLIM=bfffffff
EBP=026b0018  FLG=00010202

LIBCN0.DLL 0001:000fcc54

**Comment by AndiB-2** [comment]
I've now installed libcx-debuginfo. Please advice how to give better debug info. Thx.

**Comment by SilvanScherrer** [comment]
You need debug info for injoy as well. It on the web. And it should create a trp file.

**Comment by AndiB-2** [comment]
I've copied what I found (ijfw-dbg-os2_x86-4_2_2-GA-9174) into root of my InjoyFirewall directory. I can still reproduce the problem but I don't find any .trp file. Neither in Injoy directory nor in \usr\lib

I also found ijfw-dbg-os2_x86-4_2_2-Beta-88.zip. Guess I found it in a ticket a while a ago but not sure. Similar contents but I don't think these matches my files in my Injoy directory as they are dated 2014.

That's the latest fgui I have - 

{0}[p:\injoyfirewall4.2.2] bldlevel fgui.exe
Build Level Display Facility Version 6.12.675 Sep 25 2001
(C) Copyright IBM Corporation 1993-2001
Signature:       @#bww bitwise works GmbH:4.2.2#@##1## 22 May 2014 13:49:58
Firewall GUI
Vendor:          bww bitwise works GmbH
Revision:        4.02.2
Date/Time:       22 May 2014 13:49:58
Build Machine:   ECS26154618
File Version:    4.2.2
Description:     Firewall GUI

I now found a beta version somewhere which has the same problem. 
------------------------------------------------------------

11-10-2020  20:26:26  SYS3171  PID 0d40  TID 0004  Slot 0109
T:\TEMP\IJFW-OS2_X86-4_2_2-BETA-88\FGUI.EXE
c0000005
19facc54
P1=00000002  P2=027dfbcc  P3=XXXXXXXX  P4=XXXXXXXX  
EAX=0000150b  EBX=00000000  ECX=027e047c  EDX=027dfbcc
ESI=027e003c  EDI=f8211ccc  
DS=0053  DSACC=d0f3  DSLIM=bfffffff  
ES=0053  ESACC=d0f3  ESLIM=bfffffff  
FS=150b  FSACC=00f3  FSLIM=00000030
GS=0000  GSACC=****  GSLIM=********
CS:EIP=005b:19f00473  CSACC=d0df  CSLIM=bfffffff
SS:ESP=0053:027dfff0  SSACC=d0f3  SSLIM=bfffffff
EBP=027e0058  FLG=00010206

LIBCN0.DLL 0001:000fcc54

------------------------------------------------------------


No .trp file.

**Comment by AndiB-2** [comment]
This is the beta version bldlevel. Should have been in the comment above but instead there is some text bold formated. Weird github / Seamonkey problem.

{0}[t:\temp\ijfw-os2_x86-4_2_2-beta-88] bldlevel fgui.exe
uild Level Display Facility Version 6.12.675 Sep 25 2001
C) Copyright IBM Corporation 1993-2001
ignature:       @#bww bitwise works GmbH:4.2.2#@##1## 18 Jun 2020 13:21:50
wall GUI
endor:          bww bitwise works GmbH
evision:        4.02.2
ate/Time:       18 Jun 2020 13:21:50
uild Machine:   paperino
ile Version:    4.2.2
escription:     Firewall GUI

**Comment by SilvanScherrer** [comment]
At least the Beta should create a trp file. If you have exceptq rpm installed

**Comment by SilvanScherrer** [comment]
The older might not, as we added libcx in 2017. Before we had a own exception handling and not in fgui iirc
